dotCMS/src/main/java/com/dotmarketing/factories/MultiTreeAPIImpl.java

```diff
@@ -530,27 +530,37 @@ private void updateHTMLPageVersionTS(final String pageId) throws DotDataExceptio
 
     private void refreshPageInCache(final String pageIdentifier) throws DotDataException {
 
-        CacheLocator.getMultiTreeCache().removePageMultiTrees(pageIdentifier);
-        final Set<String> inodes = new HashSet<String>();
-        final List<ContentletVersionInfo> infos = APILocator.getVersionableAPI().findContentletVersionInfos(pageIdentifier);
-        for (ContentletVersionInfo versionInfo : infos) {
-            inodes.add(versionInfo.getWorkingInode());
+        final Set<String> personalizationSet = this.getPersonalizationsForPage(pageIdentifier);
+        personalizationSet.forEach(personalization -> CacheLocator.getMultiTreeCache()
+                .removePageMultiTrees(pageIdentifier+personalization));
+
+        final Set<String> inodeSet = new HashSet<>();
+        final List<ContentletVersionInfo> contentletVersionInfos = APILocator.getVersionableAPI()
+                .findContentletVersionInfos(pageIdentifier);
+
+        for (final ContentletVersionInfo versionInfo : contentletVersionInfos) {
+
+            inodeSet.add(versionInfo.getWorkingInode());
             if (versionInfo.getLiveInode() != null) {
-                inodes.add(versionInfo.getLiveInode());
+                inodeSet.add(versionInfo.getLiveInode());
             }
         }
+
         try {
-            List<Contentlet> contentlets = APILocator.getContentletAPIImpl().findContentlets(Lists.newArrayList(inodes));
+            final List<Contentlet> contentlets = APILocator.getContentletAPIImpl()
+                    .findContentlets(Lists.newArrayList(inodeSet));
+
+            final PageLoader pageLoader = new PageLoader();
 
-            PageLoader pageLoader = new PageLoader();
+            for (final Contentlet pageContent : contentlets) {
 
-            for (Contentlet pageContent : contentlets) {
-                IHTMLPage htmlPage = APILocator.getHTMLPageAssetAPI().fromContentlet(pageContent);
+                final IHTMLPage htmlPage = APILocator.getHTMLPageAssetAPI().fromContentlet(pageContent);
 
                 pageLoader.invalidate(htmlPage, PageMode.EDIT_MODE);
                 pageLoader.invalidate(htmlPage, PageMode.PREVIEW_MODE);
             }
         } catch (DotStateException | DotSecurityException e) {
+
             Logger.warn(MultiTreeAPIImpl.class, "unable to refresh page cache:" + e.getMessage());
```

Comment:
Include "pageIdentifier" in the error message.

Reply:
Done